sdks/python/apache_beam/io/fileio_test.py

```diff
@@ -78,6 +91,88 @@ def test_read_entire_file(self):
         read_lines.append(line)
     self.assertEqual(read_lines, lines)
 
+  def test_read_entire_file_empty(self):
+    source = fileio.TextFileSource(file_path=self.create_temp_file(''))
+    read_lines = []
+    with source.reader() as reader:
+      for line in reader:
+        read_lines.append(line)
+    self.assertEqual(read_lines, [])
+
+  def test_read_entire_file_gzip(self):
+    lines = ['First', 'Second', 'Third']
```

Comment:
Probably let's also add a read test for each compression type that reads a larger input (1k records ?)